guardrails/llm_providers.py

```diff
@@ -187,23 +304,64 @@ def _invoke_llm(
             **kwargs,
         )
 
-        # Extract string from response
-        if "function_call" in openai_response["choices"][0]["message"]:  # type: ignore
-            output = openai_response["choices"][0]["message"][  # type: ignore
-                "function_call"
-            ]["arguments"]
+        # Check if kwargs stream is passed in
+        if kwargs.get("stream", None) in [None, False]:
+            # If stream is not defined or is set to False,
+            # return default behavior
+            # Extract string from response
+            openai_response = cast(Dict[str, Any], openai_response)
```

Comment:
this got a little long - punt the functionality in each block of this condition to it's own function

Reply:
Done. Created a new func to handle this branching. Now invoke_llm is a lot cleaner! Looks better, thanks for the feedback.